### PR TITLE
Cleanup status hierarchy

### PR DESCRIPTION
**Pull request by sbusk**

### CODE REVIEW

File: whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/attrs/InetnumStatus.java [hunk 1]
```diff
@@ -16,37 +17,31 @@
...
         PARENT_STATUS.put(ALLOCATED_UNSPECIFIED, EnumSet.of(ALLOCATED_UNSPECIFIED));
-        PARENT_STATUS.put(LIR_PARTITIONED_PA, EnumSet.of(ALLOCATED_UNSPECIFIED, ALLOCATED_PA, LIR_PARTITIONED_PA, SUB_ALLOCATED_PA, EARLY_REGISTRATION));
-        PARENT_STATUS.put(LIR_PARTITIONED_PI, EnumSet.of(ALLOCATED_UNSPECIFIED, ALLOCATED_PI, LIR_PARTITIONED_PI, EARLY_REGISTRATION));
-        PARENT_STATUS.put(SUB_ALLOCATED_PA, EnumSet.of(ALLOCATED_PA, LIR_PARTITIONED_PA, SUB_ALLOCATED_PA, EARLY_REGISTRATION));
-        PARENT_STATUS.put(ASSIGNED_PA, EnumSet.of(ALLOCATED_UNSPECIFIED, ALLOCATED_PA, LIR_PARTITIONED_PA, SUB_ALLOCATED_PA, EARLY_REGISTRATION, ASSIGNED_PA));
-        PARENT_STATUS.put(ASSIGNED_ANYCAST, EnumSet.of(ALLOCATED_UNSPECIFIED, ALLOCATED_PI));
-        PARENT_STATUS.put(EARLY_REGISTRATION, EnumSet.of(ALLOCATED_UNSPECIFIED, EARLY_REGISTRATION));
-        PARENT_STATUS.put(ASSIGNED_PI, EnumSet.of(ALLOCATED_UNSPECIFIED, ALLOCATED_PI, LIR_PARTITIONED_PI, EARLY_REGISTRATION, ASSIGNED_PI));
+        PARENT_STATUS.put(LIR_PARTITIONED_PA, EnumSet.of(ALLOCATED_UNSPECIFIED, ALLOCATED_PA, LIR_PARTITIONED_PA));
```
Comment (eshryane) [inline whois-rpsl/src/main/java/net/ripe/db/whois/common/rpsl/attrs/InetnumStatus.java:40]: Should we still allow LIR-PARTITIONED PA under SUB-ALLOCATED PA (for now)?

File: whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/attrs/InetnumStatusTest.java [hunk 2]
```diff
@@ -21,54 +18,26 @@
...
-        assertThat(LIR_PARTITIONED_PA.worksWithParentStatus(SUB_ALLOCATED_PA, true), is(true));
-
-        assertThat(LIR_PARTITIONED_PI.worksWithParentStatus(EARLY_REGISTRATION, true), is(true));
-        assertThat(LIR_PARTITIONED_PI.worksWithParentStatus(ALLOCATED_UNSPECIFIED, true), is(true));
-        assertThat(LIR_PARTITIONED_PI.worksWithParentStatus(LIR_PARTITIONED_PI, true), is(true));
-        assertThat(LIR_PARTITIONED_PI.worksWithParentStatus(ALLOCATED_PI, true), is(true));
-        assertThat(LIR_PARTITIONED_PI.worksWithParentStatus(LIR_PARTITIONED_PA, true), is(false));
-
-        assertThat(ALLOCATED_PI.worksWithParentStatus(ALLOCATED_UNSPECIFIED, true), is(true));
+
```
Comment (eshryane) [inline whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/attrs/InetnumStatusTest.java:29]: Should we still allow LIR-PARTITIONED PA under LIR-PARTITIONED PA (for now) ?

Comment (eshryane) [reply whois-rpsl/src/test/java/net/ripe/db/whois/common/rpsl/attrs/InetnumStatusTest.java:29]: Also under SUB-ALLOCATED PA

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/BatchUpdatesServiceTestIntegration.java [hunk 3]
```diff
@@ -437,10 +437,15 @@ public void batch_update_using_sso_credential() {
...
-        RestTest.target(getPort(), "whois/batch/TEST")
-                .request()
-                .cookie("crowd.token_key", "valid-token")
-                .post(Entity.entity(whoisResources, MediaType.APPLICATION_JSON_TYPE), WhoisResources.class);
+        try {
+            RestTest.target(getPort(), "whois/batch/TEST")
+                    .request()
+                    .cookie("crowd.token_key", "valid-token")
+                    .post(Entity.entity(whoisResources, MediaType.APPLICATION_JSON_TYPE), WhoisResources.class);
+        } catch (BadRequestException bre) {
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rest/BatchUpdatesServiceTestIntegration.java:445]: Remove the catch? (test won't fail)

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestServiceEndToEndTest.java [hunk 4]
```diff
@@ -678,27 +677,6 @@ public void apply(String input, Deque<String> result) {
                 "<credential>PasswordCredential</credential>"));
     }
 
-    @Test
-    public void create_inetnum_with_parent_without_status_to_check_error_message_beginning_with_percent_is_handled_correctly() {
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestServiceEndToEndTest.java:682]: Should this test be removed? It's testing the error message, not the missing status

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/BatchUpdatesServiceTestIntegration.java (hunk 3 above)
Comment (sbusk) [reply whois-api/src/test/java/net/ripe/db/whois/api/rest/BatchUpdatesServiceTestIntegration.java:445]: Done, cleaned up left over test code..

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestServiceEndToEndTest.java (hunk 4 above)
Comment (sbusk) [reply whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestServiceEndToEndTest.java:682]: It was creating a situation that cannot be created and then checking against (parent has no status). All that logic has been removed as it is no longer possible to create these types of situations and none exist in the data anymore. Test is therefore obsolete.

**Pull request merged**